### PR TITLE
Reopening Image details for getPod API 

### PR DESCRIPTION
**Pull request by nirdothan-zz**
Reformatted PR, this time with 2 commits:

Code changes: 40688fa0ab5761a130d219bfcf27d10955b5eefb
Artifact generation: bc0fda94a7a6784e78f0c162d08ecef773824938

The original PR desc:
Added container image details including ImageCreationTime, ImageSize and ImageVirtualSize to the ContainerStatus struct to have them exposed by the GET /api/v1/namespaces/{namespace}/pods API.
The image details are populated using an additional docker API query to inspect the image that is returned by the existing inspectContainer API call.

cc @simon3z

### CODE REVIEW

**Comment by bgrant0607** [comment]
You have unrelated changes in this PR, such as removal of generated files. 

cc @vishh, who was on the previous discussion

File: pkg/api/types.go [hunk 1]
```diff
@@ -962,10 +962,13 @@ type ContainerStatus struct {
...
-	RestartCount int    `json:"restartCount"`
-	Image        string `json:"image"`
-	ImageID      string `json:"imageID"`
-	ContainerID  string `json:"containerID,omitempty"`
+	RestartCount      int              `json:"restartCount"`
+	ContainerID       string           `json:"containerID,omitempty"`
+	Image             string           `json:"image"`
+	ImageID           string           `json:"imageID"`
+	ImageCreationTime unversioned.Time `json:"imageCreationTime,omitempty"`
+	ImageSize         int64            `json:"imageSize,omitempty"`
```
Comment (vishh) [inline pkg/api/types.go]: I assume this value will match the actual size of the container. I don't think we should expose the details of image management through the APIs. For all we know, we might start serializing all images to help with image accounting on the nodes.

File: pkg/api/types.go [hunk 2]
```diff
@@ -962,10 +962,13 @@ type ContainerStatus struct {
...
 	// Note that this is calculated from dead containers.  But those containers are subject to
 	// garbage collection.  This value will get capped at 5 by GC.
-	RestartCount int    `json:"restartCount"`
-	Image        string `json:"image"`
-	ImageID      string `json:"imageID"`
-	ContainerID  string `json:"containerID,omitempty"`
+	RestartCount      int              `json:"restartCount"`
+	ContainerID       string           `json:"containerID,omitempty"`
+	Image             string           `json:"image"`
+	ImageID           string           `json:"imageID"`
```
Comment (vishh) [inline pkg/api/types.go]: This ID is completely dependent on the runtime. Our APIs try to be agnostic to runtimes. Why do we need this? As I said in the previous PR, cAdvisor already exposes docker image information.

**Comment by bgrant0607** [comment]
What was the original PR number?

**Comment by nirdothan** [comment]
Original PR https://github.com/kubernetes/kubernetes/pull/19695

**Comment by bgrant0607** [comment]
See also #21182

**Comment by bgrant0607** [comment]
I don't understand the purpose of ImageCreationTime.

In general, this has the same issues as the original PR.

These fields should not be defined in terms of docker inspect.

Also, the difference between ImageSize and ImageVirtualSize is not clear.

And the size field names need to include the units (e.g., Bytes).
https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md#units

**Comment by nirdothan** [comment]
If we narrow it down to one size that will be meanigful for any runtime, and include the units, would it be accepted?



cc @simon3z

**Comment by simon3z** [comment]
@nirdothan ok for me, @bgrant0607 would it be better for you?

**Comment by vishh** [comment]
FYI: We already expose Images per node via `NodeStatus`. It should be possible to look up Image `Size` based on the image name in `ContainerSpec` and [`Images`](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L2101) field in `NodeStatus`.

**Comment by vishh** [comment]
Also what is the need for exposing this information?

**Comment by vishh** [comment]
@nirdothan I'm closing this PR for now. If/when you get back to this PR, feel free to re-open it and we can continue the discussion.

**Comment by nirdothan** [comment]
Sure. Attn @simon3z

**Pull request closed without merging**